lib/modules/datasource/maven/util.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { parseUrl } from '../../../util/url';
+import {
+  checkResource,
+  checkS3Resource,
+  downloadMavenXml,
+  downloadS3Protocol,
+} from './util';
+
+describe('modules/datasource/maven/util', () => {
+  describe('downloadMavenXml', () => {
+    it('returns empty object for unsupported protocols', async () => {
+      const res = await downloadMavenXml(
+        null,
+        parseUrl('unsupported://server.com/')
+      );
+      expect(res).toEqual({});
```

Comment:
i wonder why we get an empty object instead of null. 🤔

@zharinov do you know?

Reply:
It's defined to return a `MavenXml` and uses restructuring assignment:
```
{ isCacheable, xml: mavenMetadata } = await downloadMavenXml(
```
Not sure returning null as well would affect it, though it seems more correct to do so.

Regardless, I feel refactoring this is a bit out-of-scope of this long-running PR.

Reply:
Yeah, let's figure this out later :shipit: